pkgs/development/python-modules/torchmetrics/default.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cloudpickle
+, scikit-learn
+, scikitimage
+, packaging
+, psutil
+, py-deprecate
+, pytorch
+, pytestCheckHook
+}:
+
+let
+  pname = "torchmetrics";
+  version = "0.8.1";
+in
+buildPythonPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "PyTorchLightning";
+    repo = "metrics";
+    rev = "v${version}";
+    hash = "sha256-AryEhYAeC97dO2pgHoz0Y9F//DVdX6RfCa80gI56iz4=";
+  };
+
+  propagatedBuildInputs = [
+    py-deprecate
+  ];
+
+  # Let the user bring their own instance
+  buildInputs = [
+    packaging
+    pytorch
+  ];
+
+  checkInputs = [
+    scikit-learn
+    scikitimage
+    cloudpickle
+    psutil
+  ];
+
+  nativeBuildInputs = [
+    pytestCheckHook
+  ];
+
+  disabledTestPaths = [
+    # Cyclic dependency: can remove when checkPhase gets a separate derivation
+    "integrations/test_lightning.py"
```

Comment:
We can copy this approach https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/pyparsing/default.nix

Reply:
`doCheck`, `dontUseSetuptoolsCheck`, `useSetuptoolsCheck`, `dontUsePytestCheck`, 😅